### PR TITLE
feat: add support for multiple siteIds in package.json

### LINKED ISSUE #15: Recs Templates Multiple Sites
Need to support pushing recommendations templates to multiple siteIds for implementations leveraging one codebase (typically different language / regions).

We should store related siteIds in the project `package.json` `searchspring.siteId` property. In addition to the siteId, we may need the site name for verification purposes:
```
"searchspring": {
	"siteId": {
		"do4sy1": {
			"name": "site.com.au"
		},
		"4wxokl": {
			"name": "site.hk"
		}
	},
},
```

When running the `snapfu recs sync` command locally, the templates should be synced for each siteId. If no secret key is found for a siteId, inquisitor should be used to request it and store it.

We need to store secrets in Github as well. Currently we store a single secret as `WEBSITE_SECRET_KEY` - this will still be the secret name for sites with string values in `searchspring.siteId` (in package.json). Sites using the multi-site configuration will store secrets with the siteId as part of the secret name. Eg: `do4sy1-secret-key` and `4wxokl-secret-key`. Snapfu should handle setting these via `snapfu secrets`.

### Snap Action

We will need to pass the secrets to the snap-action a bit differently to allow for multi site repositories. The easiest way would be to move to passing all of the secrets if possible (1). If that does not work out, we'd have to manually add the secrets in a map (2).

1. pass secrets in its entirety
  secrets: ${{ secrets }}

2. pass secrets in a map
  secrets:
    p9dv2z: ${{ secrets.p9dv2z_SECRET_KEY }}

To handle email recommendation image generation, the Snap Action will need to upload files to multiple CDN locations as determined by the `searchspring.siteId` contents found in package.json. We will need to verify all siteId/name/secretKey entries to ensure that information is correct prior to running the rest of the action (we currently do this for single siteId / repo name / `WEBSITE_SECRET_KEY`. This increase in checks should not impact our rate limiting much, but we should space out the requests slightly to ensure that it won't in the future.

### PR DESCRIPTION
**Pull request by dkonieczek**

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -29,6 +30,98 @@ async function parseArgumentsIntoOptions(rawArgs) {
...
+
+	const getSecretKeyFromCLI = (siteId) => {
+		const secrets = args['--secrets-ci'];
+		try {
+			if (secrets) {
+				const jsonSerializingCharacter = secrets.slice(1, 2);
+				const secretsUnserialized = secrets.split(`${jsonSerializingCharacter} "`).join('"').split(`"${jsonSerializingCharacter}}`).join('"}');
+				const secretsData = JSON.parse(secretsUnserialized);
+				const secretKey =
+					secretsData[`WEBSITE_SECRET_KEY_${siteId.toUpperCase()}`] ||
```
Comment (korgon) [inline src/cli.js]: Is this for covering all cases of potentially mis-setup manual secrets?

File: src/secret.js [hunk 2]
```diff
@@ -4,25 +4,43 @@ import inquirer from 'inquirer';
...
-	});
+	];
+
+	if (typeof options.context.searchspring.siteId === 'object') {
+		const choices = Object.keys(options.context.searchspring.siteId);
+		if (choices.length) {
+			questions.unshift({
+				type: 'list',
+				name: 'siteId',
+				message: 'Please select which siteId to add/update the secretKey for:',
```
Comment (korgon) [inline src/secret.js]: Seems a bit confusing I think to ask for a secretKey first...

File: src/secret.js [hunk 3]
```diff
@@ -53,19 +71,36 @@ export const checkSecretKey = async (options) => {
...
+				} catch (err) {
+					console.log(chalk.red(`Verification of siteId and secretKey failed for ${name}`));
+					console.log(chalk.red(err));
+					exit(1);
+				}
+
+				// prevent rate limiting
+				await wait(1111);
+			}
+		} else {
```
Comment (korgon) [inline src/secret.js:97]: Could be more DRY here by refactoring repeated code into a function or something similar.

File: src/recs.js [hunk 4]
```diff
@@ -234,26 +268,54 @@ export async function syncTemplate(options) {
...
+				} catch (err) {
+					console.log(chalk.red(`${template.details.name}`), chalk.white(`[${branchName}]`));
+					console.log(chalk.red(err));
+				}
+
+				// prevent rate limiting
+				await wait(1111);
+			}
 		}
+	} else {
```
Comment (korgon) [inline src/recs.js:299]: Could be more DRY here.

File: src/recs.js [hunk 5]
```diff
@@ -196,14 +214,30 @@ export async function removeTemplate(options) {
...
+				process.stdout.write(`archiving ${templateName} for siteId ${siteId} (${name})   `);
+				await new ConfigApi(secretKey, options.dev).archiveTemplate(payload);
+				console.log(chalk.green(`${templateName}`), chalk.white(`[${branchName}]`));
+				console.log(chalk.green('Template archived in remote.'));
+			} catch (err) {
+				console.log(chalk.red(`${templateName}`), chalk.white(`[${branchName}]`));
+				console.log(chalk.red(err));
+			}
+		}
+	} else {
```
Comment (korgon) [inline src/recs.js]: Could be more DRY here.

File: src/recs.js [hunk 6]
```diff
@@ -139,36 +139,55 @@ export async function listTemplates(options) {
...
+						const [name, branch] = template.name.split('__');
+
+						console.log(
+							`      ${template.managed ? '' : `${chalk.yellow('*')} `}${chalk.green(name)} ${branch ? `[${branch}]` : ''} ${chalk.blueBright(
+								`(https://manage.searchspring.net/management/product-recs-templates/template-version-edit?template_name=${template.name})`
+							)}`
+						);
+					});
+				}
 			} else {
```
Comment (korgon) [inline src/recs.js]: Could be DRYer here.

**Review by korgon — CHANGES_REQUESTED** [review]
Seems to be a bit of repeated code that could be cleaned up.

Perhaps we could condense logic by always expecting an array of sites (currently called multipleSites); sites not containing multiple siteIds could just push a single site in, while multi siteId configurations would push several.

File: src/cli.js (hunk 1 above)
Comment (dkonieczek) [reply src/cli.js]: After trying to create a secret manually with lowercase characters, GitHub automatically capitalizes the key. Going to remove this and always use `toUppderCase()`

File: src/secret.js (hunk 2 above)
Comment (dkonieczek) [reply src/secret.js]: This was using `unshift` so it would ask for the siteId first. I refactored this to use `when` so that it's more clear

**Review by korgon — APPROVED** [review]

**Pull request merged**